### PR TITLE
DRILL-7338: REST API calls to Drill fail due to insufficient heap memory

### PR DESCRIPTION
**Pull request by kkhatua**
This PR allows for the 85% threshold to be customizable with a value of 0 meant for disabling.

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
 public class QueryWrapper {
   private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(QueryWrapper.class);
   // Heap usage threshold/trigger to provide resiliency on web server for queries submitted via HTTP
-  private static final double HEAP_MEMORY_FAILURE_THRESHOLD = 0.85;
+  private double memoryFailureThreshold = 0.85;
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java]: There is no need to keep it as class variable, use re-assign it each time in run method.

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java [hunk 2]
```diff
@@ -101,7 +104,7 @@ public QueryResult run(final WorkManager workManager, final WebUserConnection we
         isComplete = webUserConnection.await(TimeUnit.SECONDS.toMillis(1)); //periodically timeout 1 sec to check heap
       } catch (InterruptedException e) {}
       usagePercent = getHeapUsage();
-      if (usagePercent >  HEAP_MEMORY_FAILURE_THRESHOLD) {
+      if (memoryFailureThreshold > 0 && usagePercent >  memoryFailureThreshold) {
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java]:
```suggestion
      if (memoryFailureThreshold > 0 && usagePercent > memoryFailureThreshold) {
```

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java [hunk 3]
```diff
@@ -87,6 +87,9 @@ public QueryResult run(final WorkManager workManager, final WebUserConnection we
     }
     webUserConnection.setAutoLimitRowCount(maxRows);
 
+    //Preparing for runaway memory usage
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java]:
```suggestion
    // Preparing for runaway memory usage
```

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java (hunk 1 above)
Comment (kkhatua) [reply exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java]: +1

**Comment by kkhatua** [comment]
@arina-ielchiieva applied the recommendations.

**Comment by arina-ielchiieva** [comment]
+1, please squash the commits.

**Pull request closed without merging**